### PR TITLE
Modify rule S1764 csharp and vbnet: Change scope to Main-only

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Related to https://github.com/SonarSource/sonar-dotnet/issues/4659

### CODE REVIEW

File: rules/S1764/csharp/metadata.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  
+  "scope": "Main"
```
Comment (andrei-epure-sonarsource) [inline rules/S1764/csharp/metadata.json:2]: if you change it, you should change it for other OOP languages, too, not only C#. I don't see why at least Java shouldn't have the same behavior, and possibly other languages

I'd say this is worth a dogfood post to discuss it

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S1764/csharp/metadata.json (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply rules/S1764/csharp/metadata.json:2]: I can raise awareness about this change, but other languages would need to implement this on their analyzer side and implement it as their own PR in this repo anyway.

Comment (andrei-epure-sonarsource) [reply rules/S1764/csharp/metadata.json:2]: yes please do

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**